src/library_openal.js

```diff
@@ -1594,27 +1594,674 @@ var LibraryOpenAL = {
+#endif
+    // From the programmer's guide, ALC_OUT_OF_MEMORY's meaning is
+    // overloaded here, to mean:
+    // "The specified device is invalid, or can not capture audio."
+    // This may be misleading to API users, but well...
+    AL.alcErr = 0xA005 /* ALC_OUT_OF_MEMORY */;
+    return 0; // NULL device pointer
+  },
+
+  alcCaptureCloseDevice: function(device) {
+#if OPENAL_DEBUG
+    console.error("alcCaptureCloseDevice() is not supported yet");
+#endif
+    AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+    return false;
+  },
+
+  alcCaptureStart: function(device) {
+#if OPENAL_DEBUG
+    console.error("alcCaptureStart() is not supported yet");
+#endif
+    AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+  },
+
+  alcCaptureStop: function(device) {
+#if OPENAL_DEBUG
+    console.error("alcCaptureStop() is not supported yet");
+#endif
+    AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+  },
+
+  alcCaptureSamples: function(device, buffer, num_samples) {
+#if OPENAL_DEBUG
+    console.error("alcCaptureSamples() is not supported yet");
+#endif
+    AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+  },
+
+
+  alIsEnabled: function(capability) {
+    if (!AL.currentContext) {
+#if OPENAL_DEBUG
+      console.error("alIsEnabled called without a valid context");
+#endif
+      return false;
+    }
+    // There's no defined capability for this yet - but extensions may want
+    // to use it.
+    AL.currentContext.err = 0xA002 /* AL_INVALID_ENUM */;
+    return false;
+  },
+
+
+
+
+
+  // In this section, all alGet*() functions can be implemented by casting the
+  // return value of alGetDouble().
+  // For v-suffixed variants, the spec requires that NULL destination 
+  // pointers be quietly ignored.
+  
+  getDoubleHelper: function getDoubleHelper(funcname, param) {
+     if (!AL.currentContext) {
+#if OPENAL_DEBUG
+      console.error(funcname + " called without a valid context");
+#endif
+      return 0;
+    }
+    // Right now, none of these can be set, so we directly return
+    // the values we support.
+    switch (param) {
+    case 0xC000 /* AL_DOPPLER_FACTOR */: return 1;
+    case 0xC003 /* AL_SPEED_OF_SOUND */: return 343.3;
+    case 0xD000 /* AL_DISTANCE_MODEL */: return 0 /* AL_NONE */;
+    case 0xC001 /* AL_DOPPLER_VELOCITY */:
+      Runtime.warnOnce('Getting the value for AL_DOPPLER_VELOCITY is deprecated as of OpenAL 1.1!');
+      return 1;
+    }
+    AL.currentContext.err = 0xA002 /* AL_INVALID_ENUM */;
+    return 0;
+  },
+
+  alGetDouble: function(param) {
+    return AL.getDoubleHelper("alGetDouble", param);
+  },
+
+  alGetDoublev: function(param, data) {
+    if (!data)
+      return;
+    var val = AL.getDoubleHelper("alGetDoublev", param);
+    {{{ makeSetValue('data', '0', 'val', 'double') }}};
+  },
+
+  alGetFloat: function(param) {
+    return AL.getDoubleHelper("alGetFloat", param);
+  },
+
+  alGetFloatv: function(param, data) {
+    if (!data)
```

Comment:
if the if body is on another line, need `{,}`. But for things like this, can put the return on the same line